### ISSUE TITLE
clang-tidy `bugprone-forwarding-reference-overload` false positive on `= delete;` constructor

### ISSUE THREAD

**Issue by MikeWeller** [llvm/llvm-project#88128]
The `bugprone-forwarding-reference-overload` check in clang-tidy will warn even for a `= delete;` constructor:

```cpp
// main.cpp
class Foo {
public:
  // ...

  template <typename T>
  Foo(T&&) = delete;

  // ...
};
```

```bash
clang-tidy -checks=bugprone-* main.cpp

/path/to/main.cpp:5:3: warning: constructor accepting a forwarding reference can hide the copy and move constructors [bugprone-forwarding-reference-overload]
    5 |   Foo(T&&) = delete;
      |   ^
```

In our case, we had various templated constructors and it looks like the original author decided to explicitly delete this to warn/prevent somebody making this mistake in the future.

```
$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 17.0.6
  Optimized build.
```

**Comment by MikeWeller** [comment]
If anyone can point me to the relevant source I can have a go at a fix.

**Comment by EugeneZelenko** [comment]
https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/bugprone/ForwardingReferenceOverloadCheck.h and https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/bugprone/ForwardingReferenceOverloadCheck.cpp.

**Comment by MikeWeller** [comment]
Thanks, working on a fix now.

**Comment by MikeWeller** [comment]
https://github.com/llvm/llvm-project/pull/88138
